Refactoring

maong0927 opened the pull request:
### Issue Number
Close #

### 변경사항
전체 리팩토링 

### 새로운 기능
서버로 에러 데이터를 보낼 때 call stack 에 따른 error context 추가 해서 보내게 함. 

### 작업 유형
- [x] 신규 기능 추가
- [ ] 버그 수정
- [x] 리펙토링
- [ ] 문서 업데이트

### 체크리스트
- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

maong0927 committed:
refactor: 공통 변수 전역으로 이동

1. 공통으로 쓰는 globalObject 변수 전역으로 이동

maong0927 committed:
refactor: 자주 쓰이는 타입들 types 폴더로 이동

1. dsn, message, title, contents 타입 types 로 이동시켜 재사용할 수 있게 함.

maong0927 committed:
refactor: types 모듈에 선언한 타입 적용

1. types 모듈에 선언한 타입 적용

maong0927 committed:
refactor: type 들 파일로 구분

1. 기존 index.ts 에 있던 type 들을 각 파일로 분리

maong0927 committed:
refactor: sendEvent 호출 위치 변경

1. 기존 : captureError, captureMessage 함수에서 sendEvent 함수 호출
2. 이후 : captureEvent 함수에서 sendEvent 함수 호출

1. 기존 sendEvent 를 captureError 에서 captureMessage

maong0927 committed:
fix     : 전역 객체 위치 변경

1. 기존 : 전역 객체의 위치를 함수 밖에 두었으나
2. 현재 : 동작이 되지 않아 함수 안으로 변경

maong0927 committed:
refactor: 파일 이름 변경

1. init -> sdk 로 변경
2. init 이외에 여러 함수를 같이 반환하는데 init 이라는 파일 이름은 직관적이지 않다고 느껴졌음
3. sdk 로 이름 변경

maong0927 committed:
fix     : 파일 이름 변경에 따라 import  모듈 이름 변경

1. index -> sdk

maong0927 committed:
refactor: getUserAgentInfo 메소드 변경

기존 express 버전, browser 버전 두개로 나뉘어있던 getUserAgentInfo 함수를 하나로 변경함
이유: 로직이 같아서 하나로 합치는게 더 낫다고 판단함

maong0927 committed:
refactor: 중복되어 있던 level 넣는 로직 고침

기존 두번 넣어줬었으나 한 번으로 변경

maong0927 committed:
feat    : error 발생한 line 찾는 로직 추가

maong0927 committed:
refactor: 안쓰는 모듈 import 삭제

gitdog01 reviewed (approved):
index 의 interface 들을 옮기고 , 변수등을 약간 고쳤군요.
이런 것들이 신경쓸게 많았을 텐데 고생하셨습니다.

minbert reviewed (approved):
수고하셨습니다

maong0927 committed:
fix     : ipAddress 오타 수정

1. 기존 ipAdress -> 변경 ipAddress

Eunbin-Kim commented on packages/types/src/event/index.ts:
```diff
@@ -0,0 +1,9 @@
+export * from './event';
+export * from './message';
+export * from './option';
+export * from './user';
```
user를 두번 export 하고 있습니당!

maong0927 replied:
헐 고치겠습니다 ㅎㅎㅎ ㅎㅎ

Eunbin-Kim reviewed (approved):
고생많으셨습니다~!!👍🏻

maong0927 committed:
docs    : 사용법 추가

1. setLevel, setContext, errorHandler 추가

maong0927 committed:
Publish

 - @santry/browser@1.1.0
 - @santry/core@1.2.0
 - @santry/node@1.2.0
 - @santry/types@1.2.0
 - @santry/utils@1.2.0

maong0927 committed:
fix     : export 한번으로 수정

1. 기존 user export 가 두번 되어있었음
2. 한번으로 수정
